### PR TITLE
Tegra186 platform support v1

### PR DESCRIPTION
**Pull request by vwadekar**
Support for Tegra186 SoC. Even though this branch has a lot of changes, all of them are required to complete compilation.

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by vwadekar** [comment]
@davidcunado-arm 

Can you post the issues Jenkins found?

**Comment by davidcunado-arm** [comment]
@vwadekar 
It was the static checks failure - unfortunately by the time I went to check, the build details had been purged; I'll re-run...

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by vwadekar** [comment]
Thanks. I really feel you should make the Jenkins output visible to the PR
owners.

**Comment by nmenon** [comment]
Might be nice to use something like Travisci.org. uboot community has had much benefits

**Comment by davidcunado-arm** [comment]
@vwadekar 
They are quite a few errors / warnings - probably not worthwhile copy/paste into GitHub. So alternately best to run checkpatch locally - see instructions in User Guide: https://github.com/ARM-software/arm-trusted-firmware/blob/eebb91e21a63e9e8e203217224fc5c9ef7e44ea4/docs/user-guide.md#checking-source-code-style

Looks like the main error was the use of C99 comments.
Although a warning, could you use tabs rather than spaces? That seems to be the main source of warnings.

**Comment by vwadekar** [comment]
@davidcunado-arm 

Sorry about that. I fixed all the checkpatch errors.

**Comment by vwadekar** [comment]
@davidcunado-arm 

Do you think we can move ahead with this PR?

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vwadekar 
I'll start having a look at the PR in more detail... apologies for the delay!

**Comment by vwadekar** [comment]
No worries. Thanks for the prompt reply.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

File: plat/nvidia/tegra/common/drivers/memctrl/memctrl_v2.c [hunk 1]
```diff
@@ -284,6 +284,49 @@ void tegra_memctrl_tzdram_setup(uint64_t phys_base, uint32_t size_in_bytes)
...
+	INFO("Configuring TrustZone RAM (SysRAM) Memory Carveout\n");
+
+	/* Program the base and end values */
+	tegra_mc_write_32(MC_TZRAM_BASE, (uint32_t)phys_base);
+	tegra_mc_write_32(MC_TZRAM_END, (uint32_t)tzram_end);
+
+	/* Extract the high address bits from the base/end values */
+	val = (uint32_t)(phys_base >> 32) & TZRAM_ADDR_HI_BITS_MASK;
+	val |= (((uint32_t)(tzram_end >> 32) << TZRAM_END_HI_BITS_SHIFT) &
+		TZRAM_ADDR_HI_BITS_MASK);
```
Comment (sandrine-bailleux-arm) [inline plat/nvidia/tegra/common/drivers/memctrl/memctrl_v2.c]: Coverity picked up an issue on this code:
```operator_confusion: ((uint32_t)(tzram_end >> 32) << 8) & 3 is always 0 regardless of the values of its operands. This occurs as the bitwise operand of "|".```

I agree with the tool, this line of code looks suspicious, as we're left-shifting a value by 8 bits (which introduces 8 zero-LSB) and then extracting the last 2 bits of the result. Was this intended?

**Comment by vwadekar** [comment]
Yes, our internal checks revealed that problem as well. It should be fixed
with
https://github.com/ARM-software/arm-trusted-firmware/pull/875/commits/3f4a08b832f1b1861b84e8bb03f17a6ccd4b1a04

**Comment by davidcunado-arm** [comment]
@vwadekar 
Could you split-out that commit into a separate PR? I can then review it and merge more quickly than the larger v2 PR.

**Comment by vwadekar** [comment]
Pushed the fix as
https://github.com/ARM-software/arm-trusted-firmware/pull/878

**Pull request merged**